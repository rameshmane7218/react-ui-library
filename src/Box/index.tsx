export { Box, BoxProps } from './Box';
