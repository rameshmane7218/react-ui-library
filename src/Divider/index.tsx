export { Divider, HrProps } from './Divider';
