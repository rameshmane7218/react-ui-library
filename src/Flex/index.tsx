export { Flex, FlexProps } from './Flex';
