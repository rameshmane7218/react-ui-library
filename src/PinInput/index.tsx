export { InputField, InputFieldProps } from './InputField';
export { PinInput, PinInputProps } from './PinInput';
export { PinInputExample } from './PinInputExample';
