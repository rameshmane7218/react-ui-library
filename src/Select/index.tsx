export { Select, SelectProps } from './Select';
export { Option, OptionProps } from './Option';
