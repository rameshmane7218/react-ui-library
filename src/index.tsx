export * from './Button';
export * from './Thing';
