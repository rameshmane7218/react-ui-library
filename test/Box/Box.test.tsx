import React from 'react';
import { render } from '@testing-library/react';
import { Box } from '../../src';

describe('Box Component', () => {
  it('should render without crashing', () => {
    render(
      <Box>
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry.
      </Box>
    );
  });
  it('should render all shadows variant component', () => {
    const { asFragment } = render(
      <>
        <Box shadow="xs">xs</Box>
        <Box shadow="sm">sm</Box>
        <Box shadow="md">md</Box>
        <Box shadow="lg">lg</Box>
        <Box shadow="outline">outline</Box>
        <Box shadow="none">none</Box>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should have correctly css properties', () => {
    const { getByTestId, asFragment } = render(
      <Box
        data-testid="cssBox"
        padding={'10px'}
        width={'200px'}
        border={'1px solid red'}
        rounded={'5px'}
      >
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Aperiam
        quidem, beatae blanditiis voluptatum repellendus culpa repudiandae!
        Harum impedit nesciunt quo iusto eos, ipsum possimus officia, sunt
        voluptatem aliquid, eligendi illo!
      </Box>
    );
    expect(getByTestId('cssBox')).toHaveStyle({
      width: '200px',
      border: '1px solid red',
      borderRadius: '5px',
      padding: '10px',
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
