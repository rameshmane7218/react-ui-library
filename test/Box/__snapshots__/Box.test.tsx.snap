// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component should have correctly css properties 1`] = `
<DocumentFragment>
  <div
    class="sc-gswNZR kdirRG"
    data-testid="cssBox"
    width="200px"
  >
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. Aperiam quidem, beatae blanditiis voluptatum repellendus culpa repudiandae! Harum impedit nesciunt quo iusto eos, ipsum possimus officia, sunt voluptatem aliquid, eligendi illo!
  </div>
</DocumentFragment>
`;

exports[`Box Component should render all shadows variant component 1`] = `
<DocumentFragment>
  <div
    class="sc-gswNZR gkifKp"
    width="none"
  >
    xs
  </div>
  <div
    class="sc-gswNZR jxNCQO"
    width="none"
  >
    sm
  </div>
  <div
    class="sc-gswNZR bCWdAa"
    width="none"
  >
    md
  </div>
  <div
    class="sc-gswNZR bBaSGG"
    width="none"
  >
    lg
  </div>
  <div
    class="sc-gswNZR bIXJqn"
    width="none"
  >
    outline
  </div>
  <div
    class="sc-gswNZR fnnikn"
    width="none"
  >
    none
  </div>
</DocumentFragment>
`;
